src/Language/PureScript/Errors.hs

```diff
@@ -457,11 +460,37 @@ showSuggestion suggestion = case errorSuggestion suggestion of
   Just (ErrorSuggestion x) -> x
   _ -> ""
 
+ansiColor :: ANSI.Color -> String
+ansiColor color =
+   ANSI.setSGRCode [ANSI.SetColor ANSI.Foreground ANSI.Vivid color]
+
+ansiColorReset :: String
+ansiColorReset =
+   ANSI.setSGRCode [ANSI.Reset]
+
+codeColor :: ANSI.Color
+codeColor = ANSI.Yellow
+   
+colorCode :: Bool -> String -> String
+colorCode shouldColorCode code
+  | shouldColorCode = concat [ansiColor codeColor, code, ansiColorReset]
+  | otherwise = code
+
+colorCodeBox :: Bool -> Box.Box -> Box.Box
+colorCodeBox shouldColorCode b
+  | not shouldColorCode = b
+  | shouldColorCode && Box.rows b == 1 = Box.text (ansiColor codeColor) Box.<> b `endWith` Box.text ansiColorReset
+  | otherwise = Box.hcat Box.left
+      [ Box.vcat Box.top $ replicate (Box.rows b) $ Box.text $ ansiColor codeColor
+      , b
+      , Box.vcat Box.top $ replicate (Box.rows b) $ Box.text ansiColorReset
+      ]
+
 -- |
 -- Pretty print a single error, simplifying if necessary
 --
-prettyPrintSingleError :: Bool -> Level -> Bool -> ErrorMessage -> Box.Box
-prettyPrintSingleError full level showWiki e = flip evalState defaultUnknownMap $ do
+prettyPrintSingleError :: Bool -> Bool -> Level -> Bool -> ErrorMessage -> Box.Box
+prettyPrintSingleError (colorCode &&& colorCodeBox -> (markCode, markCodeBox)) full level showWiki e = flip evalState defaultUnknownMap $ do
```

Comment:
Yes for `ViewPatterns` (one of my favorite extensions). I'll try to work on refactroing an options record. :)